### PR TITLE
[tests] add unittests for common/utils.cpp

### PR DESCRIPTION
**Pull request by wgtdkp**

### CODE REVIEW

File: src/common/utils_test.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    buf = utils::Encode<uint8_t>(0xFC);
+    REQUIRE(buf == ByteArray{0xFC});
+    utils::Encode<uint8_t>(buf, 0xFB);
+    REQUIRE(buf == ByteArray{0xFC, 0xFB});
+
+    REQUIRE(utils::Decode<int8_t>(utils::Encode<int8_t>(0xFD)) == static_cast<int8_t>(0xFD));
+
+    buf = utils::Encode<uint32_t>(0x00010203);
+    REQUIRE(buf == ByteArray{0x00, 0x01, 0x02, 0x03});
+    REQUIRE(utils::Decode<uint32_t>(buf) == 0x00010203);
```
Comment (simonlingoogle) [inline src/common/utils_test.cpp:55]: it's big endian by default? `utils::Decode` is not explicit about it though.

File: src/common/utils_test.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+TEST_CASE("hex-encoding-decoing", "[utils]")
```
Comment (simonlingoogle) [inline src/common/utils_test.cpp:58]:
```suggestion
TEST_CASE("hex-encoding-decoding", "[utils]")
```

File: src/common/utils_test.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+TEST_CASE("integer-encoding-decoding", "[utils]")
+{
+    ByteArray buf;
+
+    buf = utils::Encode<uint8_t>(0xFC);
+    REQUIRE(buf == ByteArray{0xFC});
+    utils::Encode<uint8_t>(buf, 0xFB);
+    REQUIRE(buf == ByteArray{0xFC, 0xFB});
+
+    REQUIRE(utils::Decode<int8_t>(utils::Encode<int8_t>(0xFD)) == static_cast<int8_t>(0xFD));
```
Comment (simonlingoogle) [inline src/common/utils_test.cpp:51]:
```suggestion
    REQUIRE(utils::Decode<int8_t>(utils::Encode<int8_t>(0xFD)) == static_cast<int8_t>(0xFD));
```

add more tests? 

- uint16_t ?
- uint64_t ?
- signed types (i.e. int8_t, int16_t, int32_t, int64_t)

File: src/common/utils_test.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    REQUIRE(buf == ByteArray{0x00, 0x01, 0x02, 0x03});
+    REQUIRE(utils::Decode<uint32_t>(buf) == 0x00010203);
+}
+
+TEST_CASE("hex-encoding-decoing", "[utils]")
+{
+    ByteArray   buf;
+    std::string hexStr;
+
+    hexStr = utils::Hex(ByteArray{0x00, 0x01, 0x02, 0x03});
```
Comment (simonlingoogle) [inline src/common/utils_test.cpp:63]: Include 'abcdef' ? e.x. 0xfa, 0xce

File: src/common/utils_test.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    REQUIRE(hexStr == "00010203");
+    REQUIRE(utils::Hex(buf, hexStr) == ErrorCode::kNone);
+    REQUIRE(buf == ByteArray{0x00, 0x01, 0x02, 0x03});
+}
+
+TEST_CASE("hex-negative-decoing", "[utils]")
```
Comment (simonlingoogle) [inline src/common/utils_test.cpp:69]:
```suggestion
TEST_CASE("hex-negative-decoding", "[utils]")
```

File: src/common/utils_test.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+{
+    ByteArray buf;
+
+    SECTION("decoding HEX string with odd length should fail")
```
Comment (simonlingoogle) [inline src/common/utils_test.cpp:73]: Check empty string?

File: src/common/utils_test.cpp (hunk 1 above)
Comment (wgtdkp) [reply src/common/utils_test.cpp:55]: Yes, we don't have a situation of little endian at all, so big endian is by default. I will add a comment for the `Decode/Encode` function.

File: src/common/utils_test.cpp (hunk 6 above)
Comment (wgtdkp) [reply src/common/utils_test.cpp:109]: added in test case above this one.

File: src/common/utils_test.cpp (hunk 1 above)
Comment (wgtdkp) [reply src/common/utils_test.cpp]: documentation added.

File: src/library/commissioner_impl.cpp [hunk 7]
```diff
@@ -1439,8 +1439,9 @@ Error CommissionerImpl::DecodeActiveOperationalDataset(ActiveOperationalDataset
 
     if (auto channel = tlvSet[tlv::Type::kChannel])
     {
-        dataset.mChannel.mPage   = channel->GetValue()[0];
-        dataset.mChannel.mNumber = utils::Decode<uint16_t>(&channel->GetValue()[1]);
+        const auto &value        = channel->GetValue();
```
Comment (simonlingoogle) [inline src/library/commissioner_impl.cpp:1442]: not use auto?

Comment (wgtdkp) [inline src/library/commissioner_impl.cpp:1442]:
```suggestion
        const ByteArray &value   = channel->GetValue();
```

**Pull request merged**